instance/tests/management/test_kill_zombies.py

```diff
@@ -0,0 +1,157 @@
+            "KILL_ZOMBIES_ENABLED": "",
+        },
+    )
+    @patch("instance.tasks.call_command")
+    def test_toggle_task(self, mock_command):
+        """
+        Tests that the task doesn't run unless the env variable
+        KILL_ZOMBIES_ENABLED is set and true
+        """
+        kill_zombies_periodically()
+        self.assertFalse(mock_command.called)
+
+        with patch.dict("os.environ", {"KILL_ZOMBIES_ENABLED": "true"}):
+            kill_zombies_periodically()
+            self.assertTrue(mock_command.called)
+
+    @patch("instance.tasks.call_command")
+    def test_set_env_variables(self, mock_command):
+        """
+        Tests that the task passes env variables to
+        the kill_zombies command call
+        """
+        kill_zombies_periodically()
+        mock_command.assert_called_with(
+            "kill_zombies", region="some-region"
+        )
+
+    @patch("instance.management.commands.kill_zombies.openstack_utils.get_nova_client")
+    def test_pass_variables_to_kill_zombies(self, mock_nova_client):
+        """
+        Tests that the task passes env variables to
+        the actual kill_zombies function
+        """
+        kill_zombies_periodically()
+        mock_nova_client.assert_called_with("some-region")
+
+    def test_handle_dry_run_output(self):
+        """
+        Tests that KillZombiesRunner.handle_dry_run_output
+        works as expected
+        """
+        test_runner = KillZombiesRunner()
+        no_instances = """
+        2021-02-26 14:16:12 | Starting kill_zombies
+        2021-02-26 14:16:12 | No servers found in region some-region.
+        """
+        some_instances = """
+        2021-02-26 14:16:12 | Starting kill_zombies
+        2021-02-26 14:16:12 | Found 32 unterminated servers in region some-region.
+        2021-02-26 14:16:12 | Would have terminated 32 zombies if this weren't a dry run.
+        """
+        self.assertEqual(test_runner.handle_dry_run_output(no_instances), 0)
+        self.assertEqual(test_runner.handle_dry_run_output(some_instances), 32)
+
+    @patch.dict("os.environ", {"KILL_ZOMBIES_THRESHOLD": "3"})
+    @patch.object(KillZombiesRunner, "handle_dry_run_output", return_value=4)
+    @patch.object(KillZombiesRunner, "trigger_warning")
+    @patch("instance.tasks.call_command")
+    def test_trigger_warning_if_over_threshold(
+        self,
+        mock_handle_dry_run_output,
+        mock_trigger_warning,
+        mock_command
+    ):
+        """
+        Tests that trigger_warning is called when the number
+        of zombies to delete is over the threshold
+        """
+        kill_zombies_periodically()
+        self.assertTrue(mock_handle_dry_run_output.called)
+        self.assertTrue(mock_trigger_warning.called)
+
+    @override_settings(DEFAULT_FROM_EMAIL="from@site.com")
+    @patch.dict("os.environ", {"KILL_ZOMBIES_RECIPIENT": "to@site.com"})
+    @patch("instance.tasks.call_command")
+    def test_send_warning_email(self, mock_command):
```

Comment:
@spokerman12, this test or another one should test that the email is sent to the expected recipient(s).

Reply:
True. I added the assertion on both email testcases